### ISSUE TITLE
Anchore orb cannot find custom policy_bundle.json file

### ISSUE THREAD

**Issue by paulros** [anchore/ci-tools#6]
**Summary:**
Anchore orb for circleci cannot find custom policy_bundle.json file.

**Steps to reproduse:**
Add a policy_bundle.json file to  [project_name]/.circleci/.anchore/
It's a default folder. The orb should find the file, but you can also add a parameter:
`policy_bundle_file_path: /home/circleci/project/.circleci/.anchore/policy_bundle.json`

**Expected results:** 
I used a defult policy_bundle.json file that I found [here](https://anchore.freshdesk.com/support/solutions/articles/36000074705-policy-bundles-and-evaluation)

**Actual results**
I've got this error:
```
#!/bin/bash -eo pipefail
(anchore-cli --json policy add "/home/circleci/project/.circleci/.anchore/policy_bundle.json" | jq '.policyId' | xargs anchore-cli policy activate) &> /dev/null || \
  printf "\n%s\n" "Custom policy bundle not found - /home/circleci/project/.circleci/.anchore/policy_bundle.json - using default policy bundle."
if true; then
  anchore-cli evaluate check "anchore-registry:5000/${CIRCLE_PROJECT_REPONAME}:ci" --detail
else
  (set +o pipefail; anchore-cli evaluate check "anchore-registry:5000/${CIRCLE_PROJECT_REPONAME}:ci" --detail | tee /dev/null)
fi

Custom policy bundle not found - /home/circleci/project/.circleci/.anchore/policy_bundle.json - using default policy bundle.
```

**Comments:**
My custom policy should pass all vulnerabilities expect critical, but the circleci job feils because the default policy is set to high. 

CircleCI pipeline code: 
```
version: 2.1
orbs:
  anchore: anchore/anchore-engine@1.2.1
jobs:
  local_image_scan:
    executor: anchore/anchore_engine
    steps:
      - checkout:
          path: ~/project/src/
      - run:
          name: build container
          command: docker build -t ${CIRCLE_PROJECT_REPONAME}:ci ~/project/src/
      - anchore/analyze_local_image:
          image_name: ${CIRCLE_PROJECT_REPONAME}:ci
          timeout: '500'
          policy_failure: True
          policy_bundle_file_path: /home/circleci/project/.circleci/.anchore/policy_bundle.json
      - anchore/parse_reports
workflows:
  scan_image:
    jobs:
      - local_image_scan:
          context: global
```

Path to policy_bundle.json
/home/user/project/.circleci/.anchore/policy_bundle.json

**Comment by Btodhunter** [comment]
I'm assuming your policy bundle lives in your repo under .circleci/.anchore/policy_bundle.json. If that is the case, then you'll want to point the orb to the path of your checked out out repo. In this case, try - /home/circleci/project/src/.circleci/.anchore/policy_bundle.json

**Comment by paulros** [comment]
I tried different projects and configurations, but unfortunately it it didnot work. I got the same error message.
Project tree:
```
circleci-demo/
├── build.sh
├── .circleci
│   ├── .anchore
│   │   └── policy_bundle.json
│   └── config.yml
├── config.yml
├── deploy.sh
├── Dockerfile
├── README.md

```
Error message:
```
Custom policy bundle not found - /home/circleci/project/.circleci/.anchore/policy_bundle.json - using default policy bundle.
```

Project tree:
```
circleci-demo/
├── build.sh
├── .circleci
│   └── config.yml
├── config.yml
├── deploy.sh
├── Dockerfile
├── README.md
├── src
│   └── .circleci
│       ├── .anchore
│       │   └── policy_bundle.json
│       └── config.yml
```
Error message:
```
Custom policy bundle not found - /home/circleci/project/src/.circleci/.anchore/policy_bundle.json - using default policy bundle.
```

**Comment by paulros** [comment]
Hi!
Any update on this?

I added a job to my worklflow that lists all subfolders and files

```
pwd
/home/circleci/project

ls -la ~/project
drwxrwxr-x  7 circleci circleci 4096 Apr  4 08:34 .
drwxr-xr-x 16 circleci circleci 4096 Apr  4 08:34 ..
-rwxrwxr-x  1 circleci circleci 1374 Apr  4 08:34 build.sh
drwxrwxr-x  3 circleci circleci 4096 Apr  4 08:34 .circleci
drwxrwxr-x  3 circleci circleci 4096 Apr  4 08:34 circleci-demo
-rwxrwxr-x  1 circleci circleci 6026 Apr  4 08:34 config.yml
-rwxrwxr-x  1 circleci circleci  881 Apr  4 08:34 deploy.sh
-rwxrwxr-x  1 circleci circleci  390 Apr  4 08:34 Dockerfile
drwxrwxr-x  8 circleci circleci 4096 Apr  4 08:34 .git
drwxrwxr-x  2 circleci circleci 4096 Apr  4 08:34 kube-config
-rwxrwxr-x  1 circleci circleci    4 Apr  4 08:34 lol2.txt
-rwxrwxr-x  1 circleci circleci   24 Apr  4 08:34 README.md
drwxrwxr-x  3 circleci circleci 4096 Apr  4 08:34 src
-rw-rw-r--  1 circleci circleci  792 Apr  4 08:34 whitelist.yml

ls -la ~/project/.circleci/.anchore  
drwxrwxr-x 2 circleci circleci 4096 Apr  4 08:34 .
drwxrwxr-x 3 circleci circleci 4096 Apr  4 08:34 ..
-rwxrwxr-x 1 circleci circleci 1878 Apr  4 08:34 policy_bundle.json
``` 
I tried with absolute path in _policy_bundle_file_path_, but it didn't help
`policy_bundle_file_path: '/home/circleci/project/.circleci/.anchore/policy_bundle.json'`

The image was analyzed, but it did not pass the test due to some high vulnerabilities. My custom policy bundle file was set to only stop on "critical" vuln. The orb did not see the file. I got the same error: 
`Custom policy bundle not found - /home/circleci/project/.circleci/.anchore/policy_bundle.json - using default policy bundle.`

**Comment by Btodhunter** [comment]
@paulros Can you share the policy bundle you're trying to use? That error message isn't 100% accurate in it's description. This part of the code `anchore-cli --json policy add "/home/circleci/project/.circleci/.anchore/policy_bundle.json" | jq '.policyId' | xargs anchore-cli policy activate) &> /dev/null || printf...` Just tries to add the policy bundle, and then gives the "cannot find bundle" message if the activation fails. It's possible your policy bundle isn't valid. If you can share it, I can double check that it's valid.

**Comment by paulros** [comment]
Thank you for your reply. 
I actually use a default bundle file from the example. I only changed CVE severity threshold from high to critical:

```
{
  "id": "entur_default,
  "version": "1_0",
  "name": "Entur policy bundle",
  "comment": "Entur's default bundle",
  "whitelisted_images": [],
  "blacklisted_images": [],
  "mappings": [
    { 
      "registry": "eu.gcr.io",
      "repository": "entur-system-1287",
      "image": { "type": "tag", "value": "latest" }
      "policy_ids": [ "policy1" ],
      "whitelist_ids": [ "whitelist1", "whitelist2" ]
    },
    { 
      "registry": "docker.io",
      "repository": "entur",
      "image": { "type": "tag", "value": "latest" }
      "policy_ids": [ "policy1" ],
      "whitelist_ids": [ "whitelist1" ]
    },
    { 
      "registry": "*",
      "repository": "*",
      "image": { "type": "tag", "value": "*" }
      "policy_ids": [ "policy2" ],
      "whitelist_ids": [ "whitelist1" ]
    }
  ],
  "whitelists": [],
  "policies": [
    {
      "name": "Policy1", 
      "version": "1_0",
      "comment": "Policy for critical vuln. checks", 
      "id": "policy1", 
      "rules": [
        {
          "action": "STOP", 
          "gate": "vulnerabilities", 
          "trigger": "package"
          "id": "rule1", 
          "params": [
            { "name": "package_type", "value": "all" }, 
            { "name": "severity_comparison", "value": "=" }, 
            { "name": "severity", "value": "critical" }
          ]
        }
      ]
    },
    {
      "name": "Policy2", 
      "version": "1_0",
      "comment": "Policy for high vuln. checks", 
      "id": "policy2", 
      "rules": [
        {
          "action": "STOP", 
          "gate": "vulnerabilities", 
          "trigger": "package",
          "id": "rule1", 
          "params": [
            { "name": "package_type", "value": "all" }, 
            { "name": "severity_comparison", "value": ">=" }, 
            { "name": "severity", "value": "high" }
          ]
        }
      ]
    }
  ]
}
```

**Comment by denismakogon** [comment]
I can confirm that `policy_bundle_file_path` works as expected, the only issue about the following script:
```bash
#!/bin/bash -eo pipefail
(anchore-cli --json policy add "/home/circleci/project/.circleci/.anchore/policy_bundle.json" | jq '.policyId' | xargs anchore-cli policy activate) &> /dev/null || \
  printf "\n%s\n" "Custom policy bundle not found - /home/circleci/project/.circleci/.anchore/policy_bundle.json - using default policy bundle."
if true; then
  anchore-cli evaluate check "anchore-registry:5000/${CIRCLE_PROJECT_REPONAME}:ci" --detail
else
  (set +o pipefail; anchore-cli evaluate check "anchore-registry:5000/${CIRCLE_PROJECT_REPONAME}:ci" --detail | tee /dev/null)
fi
```
is that it's not verbose on errors which is not quite good because the UX faces certain regression and makes debugging quite hard (you'd need eventually install Anchore locally and debug each exec on your machine).

**Comment by Btodhunter** [comment]
I have changed the implementation of the policy evaluation step of the job in the most recently published orb v1.3.0. It should now give more verbose errors when there is an issue loading the bundle or when the bundle file cannot be found.

**Comment by denismakogon** [comment]
Thanks, I’ll give it a shot later today.
